### PR TITLE
Stats Admin: Add module settings api

### LINKED ISSUE #31228: Odyssey: support modules / settings / notices API forwarding
- [x] https://github.com/Automattic/jetpack/pull/31231
- [x] https://github.com/Automattic/jetpack/pull/31230
- [x] https://github.com/Automattic/jetpack/pull/31261

### PR DESCRIPTION
**Pull request by kangzj**
Partially Fixes #31228 

## Proposed changes:

The PR proposes to add API to get modules settings on Stats Dashboard, and API to update module settings. The APIs are forwarded to WPCOM.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
n/a

## Does this pull request change what data or activity we track or use?
no

## Testing instructions:

* Build Odyssey `STATS_PACKAGE_PATH=/path/to/jetpack/projects/packages/stats-admin yarn dev`
* Open `/wp-admin/admin.php?page=stats&flags=stats/highlights-settings`
* Change settings of the Highlight module
* Refresh page
* Ensure settings is persisted

image

### CODE REVIEW

File: projects/packages/stats-admin/src/class-rest-controller.php [hunk 1]
```diff
@@ -563,6 +584,57 @@ public function unmark_referrer_spam( $req ) {
...
+					$req->get_params()
+				)
+			),
+			'v2',
+			array(
+				'timeout' => 5,
+			),
+			null,
+			'wpcom',
+			false,
```
Comment (kangzj) [inline projects/packages/stats-admin/src/class-rest-controller.php]:
```suggestion
			true,
```

**Review by dognose24 — DISMISSED** [review]
Works as described! 👍🏼

**Review by jsnmoon — COMMENTED** [review]
What happens if the user is on the latest Odyssey Stats alongside an older Jetpack plugin lacking these local endpoints? Would they be unable to persist module settings?

**Comment by kangzj** [comment]
> What happens if the user is on the latest Odyssey Stats alongside an older Jetpack plugin lacking these local endpoints? Would they be unable to persist module settings?

@jsnmoon  For site that are still on old Jetpack, we'd use default settings, and hide the settings interface if the backend dosn't support it. The task could be found here:

https://github.com/Automattic/wp-calypso/issues/77863

**Review by dognose24 — APPROVED** [review]
LGTM! 👍🏼

**Review by dognose24 — APPROVED** [review]
LGTM! 👍🏼

**Pull request merged**